### PR TITLE
Suppot EQUIL + WATER + GAS

### PR DESCRIPTION
**Pull request by totto82**
Seems like most of the work was already done. @bska Did I miss something? Only tested on a simple case without capillary pressure. So definitely it needs more testing.

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
When testing the gas+water initialization I noticed that the results form the gas+oil initialization seemed wrong. (I think it used the wrong cap-pressure for the endpoints). 

To fix this I changed `accountForScaledSaturations` which means that this PR will change the behavior for more than the water + gas case. I may have missed something (or misunderstood somethings) so comments are welcome.

**Comment by totto82** [comment]
jenkins build this please

**Comment by bska** [comment]
jenkins build this please

**Comment by totto82** [comment]
sorry for updating this again and again. I was editing the data file in the build directory... Now the equil test should not fail...

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
> To fix this I changed accountForScaledSaturations which means that this PR will change the behavior for more than the water + gas case. I may have missed something (or misunderstood somethings) so comments are welcome.

After fixing the fix that fixed the 2p case also the 3p case is working and all tests are fine.

**Comment by plgbrts** [comment]
I ran a two phase gas-water case with nonzero capillary pressure with the EQUIL keyword and compared Flow with E100.
Flow:
image
E100:
image

The behavior is quite different but could not find an inconsistency in the code. Maybe good to have a similar comparison for an oil-gas case to see what is to be expected? 

The gas water case used for the comparison: 
[SPE1CASE2_GASWATER_EQUIL.zip](https://github.com/OPM/opm-simulators/files/10403738/SPE1CASE2_GASWATER_EQUIL.zip)

File: ebos/equil/initstateequil.cc [hunk 1]
```diff
@@ -732,75 +741,106 @@ accountForScaledSaturations()
...
+        .oilWaterScaledEpsInfoDrainage(this->evalPt_.position->cell);
 
     const auto thresholdSat = 1.0e-6;
     if (watActive && ((sw + thresholdSat) > scaledDrainageInfo.Swu)) {
         // Water saturation exceeds maximum possible value.  Reset oil phase
         // pressure to that which corresponds to maximum possible water
         // saturation value.
         this->fluidState_.setSaturation(this->waterPos(), scaledDrainageInfo.Swu);
+        if (oilActive) {
+            this->fluidState_.setSaturation(this->oilPos(), so - sw + scaledDrainageInfo.Swu);
```
Comment (bska) [inline ebos/equil/initstateequil.cc:764]: How does this correctly adjust the oil saturation?  Suppose, for example, that inverting the capillary pressure curves gives
```
(Sw, So, Sg) = (0.4, 0.5, 0.1)
```
but that `Swu = 0.3`.  If we then reduce `Sw` to `Swu`, then I would **typically** expect the surplus 0.1 to be distributed among the other two phases, and if we assign the surplus entirely to the oil, then we **should** have
```
(Sw, So, Sg) = (0.3, 0.6, 0.1)
```
On the other hand, this expression gives
```
So = 0.5 - 0.4 + 0.3 = 0.4
```
from which we get the saturation distribution
```
(Sw, So, Sg) = (0.3, 0.4, 0.1)
```
which does not sum to one instead.  What am I missing?

**Review by bska — COMMENTED** [review]
This looks good to me for the most part, except for the saturation updates in `accountForScaledSaturations()` which I don't fully comprehend.  I also think we should rename one of the intermediate variables to reflect the common case.

**Comment by totto82** [comment]
>The behavior is quite different but could not find an inconsistency in the code. Maybe good to have a similar comparison for an oil-gas case to see what is to be expected?

Do we compare two different pressures? OPM Flow will output the water phase pressure for a gas-water case. The pressure shown for OPM seems to be 4700 - pc(sw). Have capillary pressure + gas + water, with explicit initialization been tested? This may not be related to the initialization.

File: ebos/equil/initstateequil.cc (hunk 1 above)
Comment (totto82) [reply ebos/equil/initstateequil.cc]: > What am I missing?

Just that I forgot to flip signs. I am surprised that this bug in the PR did not affect any of the tests.

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
With #4372  The pressure is similar to the E100 results @plgbrts showed.

**Review by bska — APPROVED** [review]
Thanks a lot for the updates.  I'll merge this into the master branch now.

File: ebos/equil/initstateequil.cc (hunk 1 above)
Comment (bska) [reply ebos/equil/initstateequil.cc]: > I am surprised that this bug in the PR did not affect any of the tests.

I am too.  On the other hand, It's **possible** that the tests don't need this kind of adjustment at all, and that if they actually do then they mostly care about the water saturation and implicitly assume `So = 1 - Sw`.

**Pull request merged**